Dockerfile

```
@@ -2,21 +2,24 @@ FROM node:19-alpine AS BUILD_IMAGE
 
 WORKDIR /usr/src/app
 
-COPY yarn.lock package.json tsconfig.json .
+COPY pnpm-lock.yaml package.json tsconfig.json .
 COPY src/ ./src
 COPY .swcrc ./.swcrc
 
-RUN yarn && yarn build && rm -rf node_modules && yarn --production
+RUN npm install -g pnpm
+RUN pnpm install && pnpm build && rm -rf node_modules && pnpm install --prod
```

```suggestion
RUN pnpm install --frozen-lockfile \
  && pnpm build \
  && pnpm install --frozen-lockfile --prod
```

Frozen lockfile ensure nothing gets changed from what is specified. Also running `pnpm` with `--prod` will automatically remove dev modules from node_modules, so it's not a requirement to delete it first.

---

Oooh, that's a nice bonus! Nice catch 🔥